django/db/models/expressions.py

```diff
@@ -1156,7 +1156,7 @@ def as_sqlite(self, compiler, connection):
     def as_mysql(self, compiler, connection):
         template = None
         if self.nulls_last:
-            template = 'IF(ISNULL(%(expression)s),1,0), %(expression)s %(ordering)s '
+            template = 'ISNULL(%(expression)s), %(expression)s %(ordering)s '
         elif self.nulls_first:
             template = 'IF(ISNULL(%(expression)s),0,1), %(expression)s %(ordering)s '
```

Comment:
That would also mean that this could be `%(expression)s IS NOT NULL`, getting rid of the `IF()` here too...